lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -453,6 +460,68 @@ void SPIRVRegularizeLLVMBase::lowerFuncPtr(Module *M) {
     lowerFuncPtr(I.first, I.second);
 }
 
+void SPIRVRegularizeLLVMBase::addKernelEntryPoint(Module *M) {
+  std::vector<Function *> Work;
+
+  // Get a list of all functions that have SPIR kernel calling conv
+  for (auto &F : *M) {
+    if (F.getCallingConv() != CallingConv::SPIR_KERNEL)
+      continue;
+
+    Work.push_back(&F);
+  }
+  for (auto &F: Work) {
+
+    // for declarations just make them into SPIR functions.
+    if (F->empty()) {
+      F->setCallingConv(CallingConv::SPIR_FUNC);
+      continue;
+    }
+
+    // Otherwise add a wrapper around the function to act as an entry point.
+    FunctionType *FType = F->getFunctionType();
+    std::string wrap_name = "__spirv_entry_" + static_cast<std::string>(F->getName());
```

Comment:
It would be good to create const with "__spirv_entry_" value and use it instead of hard code.

Reply:
done